server/src/main.rs

```diff
@@ -503,80 +620,114 @@ fn debug_execution_trace(mut req: Request, mut res: Response, caps: Captures, ch
     let execution = caps[2].parse::<u64>().unwrap();
     io::copy(&mut req, &mut io::sink()).unwrap();
 
-    // TODO: this is a hack for the prototype; implement process executions
-    if execution == 0 {
-        use serde_json::{Value, Map};
+    let mut error = false;
+    let mut child = child.lock().unwrap();
+    {    
```

Comment:
This is an awful lot of code to put in a block- should do as little as possible inside it so the rest of the function can stay at its old indentation. Could call `child.as_mut()` in multiple places when necessary. This should let us eliminate the `error` flag here as well.